moveit_ros/planning/moveit_cpp/src/planning_component.cpp

```diff
@@ -189,7 +216,127 @@ PlanningComponent::PlanSolution PlanningComponent::plan(const PlanRequestParamet
   //    visual_tools_->publishRobotState(last_solution_trajectory_->getLastWayPoint(), rviz_visual_tools::TRANSLUCENT);
   //  }
   //}
-  return *last_plan_solution_;
+
+  if (update_last_solution)
+  {
+    last_plan_solution_ = plan_solution;
+  }
+  return *plan_solution;
+}
+
+PlanningComponent::PlanSolution PlanningComponent::plan(const MultiPipelinePlanRequestParameters& parameters,
+                                                        SolutionCallbackFunction solution_selection_callback,
+                                                        StoppingCriterionFunction stopping_criterion_callback)
+{
+  // Create solutions container
+  PlanningComponent::PlanSolutions planning_solutions{ parameters.multi_plan_request_parameters.size() };
+  std::vector<std::thread> planning_threads;
+  planning_threads.reserve(parameters.multi_plan_request_parameters.size());
```

Comment:
Wouldn't it make sense to sanity-check the requested number of threads, especially since `planning_attempts` is also handled as thread count by OMPL?

Reply:
sanity check against `std::thread::hardware_concurrency()` you mean? That should at most produce a warning. Even with a single core there might be value in running multiple threads concurrently. Also, `std::thread::hardware_concurrency()` can (apparently) return 0 on some platforms.

Reply:
I've added a warning for it.